iothub/service/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -15,24 +15,11 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public class IotHubConnectionStringBuilder
     {
-        private const char ValuePairDelimiter = ';';
-        private const char ValuePairSeparator = '=';
-        private const string HostNameSeparator = ".";
-
-        private const string HostNamePropertyName = "HostName";
-        private const string SharedAccessKeyNamePropertyName = "SharedAccessKeyName";
-        private const string SharedAccessKeyPropertyName = "SharedAccessKey";
-        private const string SharedAccessSignaturePropertyName = "SharedAccessSignature";
-        private const string DeviceIdPropertyName = "DeviceId";
-        private const string ModuleIdPropertyName = "ModuleId";
-        private const string GatewayHostNamePropertyName = "GatewayHostName";
-
         private static readonly TimeSpan s_regexTimeoutMilliseconds = TimeSpan.FromMilliseconds(500);
-        private static readonly Regex s_hostNameRegex = new Regex(@"[a-zA-Z0-9_\-\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, s_regexTimeoutMilliseconds);
-        private static readonly Regex s_sharedAccessKeyNameRegex = new Regex(@"^[a-zA-Z0-9_\-@\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, s_regexTimeoutMilliseconds);
-        private static readonly Regex s_sharedAccessKeyRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, s_regexTimeoutMilliseconds);
-        private static readonly Regex s_sharedAccessSignatureRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, s_regexTimeoutMilliseconds);
-        private static readonly Regex s_idRegex = new Regex(@"^[A-Za-z0-9\-:.+%_#*?!(),=@;$']{1,128}$", RegexOptions.Compiled | RegexOptions.IgnoreCase, s_regexTimeoutMilliseconds);
+        private static readonly Regex s_hostNameRegex = new(@"[a-zA-Z0-9_\-\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, s_regexTimeoutMilliseconds);
```

Comment:
I'd like us to not own any such validations. We can delegate that responsibility to service directly.
If we add validations like this in our client code, we'll need to keep updating them as and when service choses to update them.